### ISSUE TITLE
Surveyor sheets not building correctly

### ISSUE THREAD

**Issue by dbl3raf** [bird-team/brisbane-bird-atlas#118]
There's an alignment problem in the surveyor sheets that wasn't there before:
![image](https://user-images.githubusercontent.com/23551540/52905059-318c6d00-3280-11e9-9c09-05146a28dd50.png)

**Comment by jeffreyhanson** [comment]
Sorry for the slow response - did you add or remove any new species to the atlas with the new data?

**Comment by dbl3raf** [comment]
yes I did - my latest thought is that the name of "Mallard x Pacific Black Duck (hybrid)" is too long, and has pushed the columns to the right. I haven't investigate it yet, but for now @jeffreyhanson I think there's no need for you to investigate the code - I'll just make the English name shorter

**Comment by jeffreyhanson** [comment]
Ok, sounds good. Also, you could try playing around with the spacing in the surveyor sheet checklists using the  `grid_checklist_left_margin` and `grid_checklist_right_margin` parameters in the parameters YAML file (https://github.com/bird-team/brisbane-bird-atlas/blob/master/data/parameters/parameters.yaml)

**Comment by Louis-Backstrom** [comment]
This is hopefully fixed in [53b4c1b](https://github.com/bird-team/brisbane-bird-atlas/commit/53b4c1b8af92124e18c5bcd11aa1b38ee13e123b)

**Comment by Louis-Backstrom** [comment]
This has been successfully fixed thanks to the above commit with the rebuilding of the assets. Now I just need to get the assets back up...

**Comment by Louis-Backstrom** [comment]
I'm going to reopen this issue as I've created new formatting problems with the latest changes to the species list. This is twofold:

(1) I've added a whole bunch of new species (now that #129 is fixed) and domestics etc, so the old 2 pages/3 columns layout is now insufficent. I've "fixed" this by adding another page to the checklist (in a189fc0) but it needs tweaking 

(2) The new species have very long names (courtesy of "Domestic type") which are causing the same grief as the original issue raised here. I'm going to try and sort out the formatting to fix this, but I'd like to keep the names long if possible (especially as we have species like Cockatiel, Mallard etc which have both domestic and wild types recorded and in the Atlas).

I will update this issue with my progress, although it'll probably be quite slow as it takes a long time to rebuild the assets each time...

If @jeffreyhanson has any ideas on how to make the checklist formatting more functional I'm all ears too.

**Comment by Louis-Backstrom** [comment]
I think for the checklist, a restructuring is needed.

I estimate that once done we're going to have ~450 taxa all up in the Atlas (all normal species, escapes, domestics, etc.). I think therefore, 3 pages of 3 columns of 50 species in each column is probably going to work well. I'm not entirely sure how to achieve this - I haven't been able to find in the code (and I've looked!) where the option for changing the font size within the checklist is - as I think we need to make it smaller to accommodate the longer species names.

Currently trying to troubleshoot a few things that will hopefully fix things up a bit, then I'll try and push the fixed sheets to the storage. This is probably going to be a big task that needs addressing over the next few months at periodic intervals, as we add new species to the Atlas, play around with things etc.

**Comment by dbl3raf** [comment]
good idea to plan ahead and build in the space we are likely to need

**Comment by Louis-Backstrom** [comment]
Right, have found the parts of the code I need to play around with and am currently working on finding the right balance. Once I've done this I think I'll add as many of the remaining taxa to the atlas as I can (inclusive of hybrids, domestics, escapees etc) and hopefully the surveyor sheets will start looking good.

**Comment by Louis-Backstrom** [comment]
Okay, getting toward a nice end product. The final finnicky thing I've got to work out at the moment is how to make the seasonal columns each the same size. Based off my reading of the code, the best way to do this is to make the text at the top of each column the same width, which I should be able to achieve by monospacing it (so "Su", "Au", "Wi", "Sp" are all the same width). However, I've got no idea actually how to do this :( - @jeffreyhanson are you able to help?

This is obviously a very minor thing, and not a big issue. Currently somewhat working around it by increasing the text to 3 letters, which averages things out a bit (but still is somewhat uneven). Have sorted the spacing, and have got it so we will have enough room for 450 entries (50 in each column) - I'll try to add everything we've currently got confirmed in the Atlas soon.

**Comment by dbl3raf** [comment]
Maybe could use single letters S, A, W, S, which might also create more room by thinning the columns even further. Not too much ambiguity given the order of the seasons? Or is there?

**Comment by Louis-Backstrom** [comment]
I did consider that, but W is quite a wide letter so it's more or less the same issue. Plus I don't think we need to narrow the columns any further than they currently are. Trying to get a full refresh of the assets done so I can upload them, but no dice still.

**Comment by Louis-Backstrom** [comment]
Alright, assets successfully built. @dbl3raf can you have a look at the formatting and verify that it looks alright?

**Comment by dbl3raf** [comment]
Surveyor sheets look great, nice job @Louis-Backstrom. Two points:

In the Mansfield one, I notice it now wraps onto another page - not has something changed about the map image alignment, as there seems to be a bigger gap now between the map and text that comes below it.

There only seems to be six spaces left for new additions? Or are there more spaces available at the foot of the table on each page? I guess an allied question is how easy would it be to narrow the height of the rows a bit if needed when Ged has finished finding half a dozen more firsts for Brisbane!

**Comment by Louis-Backstrom** [comment]
Mansfield one is too long on the description I think, so that is why it's pushed it to a new page. However, I was messing around with various spacing values so _might_ have increased the spacing there, which is causing additional issues. If so, that'll be an easy fix once I work out which value it is. I'll have a look when I get home tonight and see if the spacing has changed.

As for the table, in theory there's room for ~6 more rows on each page, so 60 more taxa all up with the extra space on the last page. My understanding from the code is that it calculates the total number of taxa, divides it by 9 (rounding up) and has that many rows on each page, rather than filling each page up before going onto the next - @jeffreyhanson wrote the original code so no doubt understands it better. Narrowing the row height is easy (one of the many things I worked out how to do yesterday) - for memory I've got it on one-and-a-half spacing at the moment, but it was previously on single spacing and I believe we can change it to essentially any value we want (so can make it 1.45 or similar to add a 7th extra row, should we need it). However, 60 new taxa should give us plenty of room to work with - even if all of the pending taxa on the working list get accepted (which is not likely), that's still only 30 or so more. Going at a rate of ~5 new species per year on eBird, I reckon we'll be fine for a while yet.

**Comment by Louis-Backstrom** [comment]
As a side note, any ideas for what to do with New Holland X white cheeked? I've currently compressed it down into the very dodgy "Holland cheeked" Honeyeater - the full name is too long, even with smaller font size (another change from yesterday)

**Comment by Louis-Backstrom** [comment]
> In the Mansfield one, I notice it now wraps onto another page - not has something changed about the map image alignment, as there seems to be a bigger gap now between the map and text that comes below it.
> 

Okay, just got home and had a look at an old copy of the sheets vs the current - nothing on the first page should have changed, so any pagination problems are just the result of the text being too long and pushing the table onto the next page. That being said, I agree that the description seems an unnecessarily long way spaced from the map - I might try and work out how to change that if I've got time in the next few days. The only spacing change that has made it through to this version is I've reduced the space between the header and the start of the species checklist:

![image](https://user-images.githubusercontent.com/44354406/55682191-55f8f180-5973-11e9-955f-8b12afb153b7.png)
![image](https://user-images.githubusercontent.com/44354406/55682195-5e512c80-5973-11e9-8844-d03923a67d58.png)

I don't think it's too tight now but I might revert it back so there's a little more room up the top. Other than that, and the naming of the honeyeater hybrid I think this issue is more or less fine for the time being?
